server/src/main/java/org/elasticsearch/index/analysis/Analysis.java

```diff
@@ -83,6 +84,8 @@ public class Analysis {
     private static final DeprecationLogger DEPRECATION_LOGGER = DeprecationLogger.getLogger(Analysis.class);
     private static final Logger logger = LogManager.getLogger(Analysis.class);
 
+    public static final NodeFeature PERSIAN_ANALYZER_WITH_STEMMER = new NodeFeature("persian_analyzer_with_stemmer");
```

Comment:
I originally wanted to put this into the PersianAnalyzerProvider but I cannot see analysis-common from RestFeatures where I need to reference it. Still not sure if its overkill to add a "capability" for this but saw it also done around modifications in Highlighting (see RestFeatures) so I think its okay. Let me know if you know better options.

Reply:
The purpose of these is to distinguish between node deployments. Since we know this will be deployed with Lucene 10 changes, let's just have a "Lucene 10 upgrade" feature and use it. 

We can also use that lucene 10 feature for any other bwc things that need changing.

Reply:
Moved to `SearchFeatures`